packages/headless/doc-parser/src/entity-builder.ts

```diff
@@ -0,0 +1,100 @@
+import {Parameter} from '@microsoft/api-extractor-model';
+import {DocComment, DocNode, DocNodeKind} from '@microsoft/tsdoc';
+import {AnyEntity, Entity, FuncEntity, ObjEntity} from './entity';
+import {emitAsTsDoc} from './tsdoc-emitter';
+
+interface Comment {
+  comment: DocComment | undefined;
+}
+
+interface EntityOptions extends Comment {
+  name: string;
+  type: string;
+  isOptional: boolean;
+}
+
+interface FuncEntityOptions extends Comment {
+  name: string;
+  params: AnyEntity[];
+  returnType: string | ObjEntity | FuncEntity;
+}
+
+export function buildEntity(config: EntityOptions): Entity {
+  const type = sanitizeType(config.type);
+  const desc = getSummary(config.comment);
+
+  return {
+    name: config.name,
+    isOptional: config.isOptional,
+    type,
+    desc,
+  };
+}
+
+export function buildParamEntity(param: Parameter): Entity {
+  const desc = getParamDescription(param);
+  const type = sanitizeType(param.parameterTypeExcerpt.text);
+
+  return {
+    name: param.name,
+    isOptional: false,
+    desc,
+    type,
+  };
+}
+
+export function buildFuncEntity(config: FuncEntityOptions): FuncEntity {
+  const desc = getSummary(config.comment);
+
+  return {
+    name: config.name,
+    params: config.params,
+    returnType: config.returnType,
+    desc,
+  };
+}
+
+function sanitizeType(type: string) {
+  return type.replace(/\$[0-9]{1}/, '');
+}
+
+function getSummary(comment: DocComment | undefined) {
+  if (!comment) {
+    return '';
+  }
+
+  try {
+    const nodes = removeBlockTagNodes(comment.summarySection.nodes);
+    return emitAsTsDoc(nodes);
+  } catch (e) {
+    console.log(
+      'failed to get summary for comment:\n',
+      comment.emitAsTsdoc(),
+      e
+    );
+    return '';
+  }
+}
+
+function getParamDescription(param: Parameter) {
+  const nodes = param.tsdocParamBlock?.content.getChildNodes() || [];
+
+  try {
+    return emitAsTsDoc((nodes as unknown) as readonly DocNode[]);
+  } catch (e) {
+    console.log('failed to description for param:', param.name, e);
+    return '';
+  }
+}
+
+function removeBlockTagNodes(nodes: readonly DocNode[]) {
```

Comment:
This is a work-around to support custom tags.

Atm, we are using `@default` which is not standard (should be `@defaultValue`). Why not switch?

VSCode applies special formatting to only `@default` annotations.
We are also using other custom tags such as `@minimum` which have no spec-compliant counterpart.

The true fix would be to[ configure the custom tags](https://tsdoc.org/pages/packages/tsdoc-config/) on the TSDoc parser. However, we are not invoking the parser directly, rather using it through `api-extractor-model`, and I did not see a way to configure the TSDoc parser through it.

I took a note to revisit this,

Reply:
Yeah we should switch to defaultValue, I didn't know default wasn't the default 😏 
Could we have a vscode config in the project which highlights some tags such as defaultValue, is that a thing?

Reply:
I haven't seen such config, but even if it was possible, every headless user would need to set that config which is not ideal.